.github/workflows/tests.yml

```diff
@@ -39,3 +39,72 @@ jobs:
         npm rebuild node-sass
     - name: Run tests
       run: yarn test
+
+  visual_tests:
+    name: Frontend Visual Tests
+    needs: test
+    if: ${{ github.event_name == 'pull_request' && needs.test.result == 'success' }}
+    runs-on: ubuntu-latest
+    outputs:
+      percy_url: ${{ steps.extract-url.outputs.percy_url }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18.x'
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+      - name: Install dependencies
+        run: |
+          yarn --frozen-lockfile
+          npm rebuild node-sass
+      - name: Start server
```

Comment:
Could we avoid this step if we use `test:visual` for running the visual tests? With that command we already run the server and in the jest config we already wait for it to be running

Reply:
The reason for me to use this was to generate the `jsdocs` and `environment.js` files so that the server can be started. The `test:visual` command is set to use the `yarn dev-only` command for starting the server as it's fast, but it doesn't generate those files, which are required for the server to run properly.

Reply:
I think we can run `yarn pregenerate` (or the needed generate commands that are pre-requisites) and then `yarn test:visual`, so we dont have the logic of waiting for the server to be running in the workflow. `yarn dev` start some other processes also as the lint watch and api watch that we probably dont need to run here.

Reply:
I've made the required changes :)